File: e2e/cypress-javascript/interactions/pages/MainPage.js
class MainPage {
  get taskRows() {
    return cy.get('[data-testid^="task-title-"]');
  }

  get createTaskButton() {
    return cy.get('[data-testid="add-task-button"]');
  }

  get languageSwitcher() {
    return cy.get('[data-testid="language-switcher"]');
  }

  get tableHeaders() {
    return cy.get('.p-datatable-thead');
  }

  get taskTableBody() {
    return cy.get('.p-datatable-tbody');
  }

  taskRowByTitle(title) {
    return cy.get('[data-testid^="task-title-"]').contains(title);
  }

  taskInfoButtonById(id) {
    return cy.get(`[data-testid="info-button-${id}"]`);
  }

  taskEditButtonById(id) {
    return cy.get(`[data-testid="edit-button-${id}"]`);
  }

  taskDeleteButtonById(id) {
    return cy.get(`[data-testid="delete-button-${id}"]`);
  }

  taskTitleByTitle(title) {
    return cy.get('[data-testid^="task-title-"]').contains(title);
  }

  statusTag(status) {
    return cy.get(`[data-testid="status-tag-${status}"]`).first();
  }

  priorityTag(priority) {
    return cy.get(`[data-testid="priority-tag-${priority}"]`).first();
  }
}

module.exports = { MainPage };

File: e2e/capybara-ruby/interactions/pages/main_page.rb
# frozen_string_literal: true

require_relative '../../support/capybara_helpers'

class MainPage
  include CapybaraHelpers
  def task_rows
    find_all('[data-testid^="task-title-"]')
  end

  def create_task_button
    find('[data-testid="add-task-button"]')
  end

  def language_switcher
    find('[data-testid="language-switcher"]')
  end

  def table_headers
    find('.p-datatable-thead')
  end

  def task_table_body
    find('.p-datatable-tbody')
  end

  def task_row_by_title(title)
    find('[data-testid^="task-title-"]', text: title)
  end

  def task_info_button_by_id(id)
    find("[data-testid=\"info-button-#{id}\"]")
  end

  def task_edit_button_by_id(id)
    find("[data-testid=\"edit-button-#{id}\"]")
  end

  def task_delete_button_by_id(id)
    find("[data-testid=\"delete-button-#{id}\"]")
  end

  def task_title_by_title(title)
    find('[data-testid^="task-title-"]', text: title)
  end

  def status_tag(status)
    first("[data-testid=\"status-tag-#{status}\"]")
  end

  def priority_tag(priority)
    first("[data-testid=\"priority-tag-#{priority}\"]")
  end
end
